plugin.rb

```diff
@@ -217,40 +194,28 @@ def can_vote?
       end
 
       def vote_count
-        if count = self.custom_fields[DiscourseVoting::VOTE_COUNT]
-          # we may have a weird array here, don't explode
-          # need to fix core to enforce types on fields
-          count.try(:to_i) || 0
-        else
-          0 if self.can_vote?
-        end
+        self.vote_counter&.counter.to_i
       end
 
-      def user_voted(user)
-        if user && user.custom_fields[DiscourseVoting::VOTES]
-          user.custom_fields[DiscourseVoting::VOTES].include? self.id
+      def user_voted?(user)
```

Comment:
Since we now have a `TopicExtension` for votes, maybe all these voting related methods should fall under there as well?